activerecord/lib/active_record/migration.rb

```diff
@@ -1013,10 +1033,15 @@ def record_version_state_after_migrating(version)
         ActiveRecord::SchemaMigration.where(:version => version.to_s).delete_all
       else
         migrated << version
-        ActiveRecord::SchemaMigration.create!(:version => version.to_s)
+        ActiveRecord::SchemaMigration.create!(version: version.to_s, environment: ActiveRecord::ConnectionHandling::DEFAULT_ENV.call)
       end
     end
 
+    def self.production?
+      raise NoEnvironmentInSchemaError unless SchemaMigration.environment_is_stored?
+      ActiveRecord::SchemaMigration.order(:created_at).last.environment == "production"
```

Comment:
This won't work. When using `db/structure.sql` you'll get the records from that table in your dumped output like so:

```sql
INSERT INTO schema_migrations (version) VALUES ('20150422072614');

INSERT INTO schema_migrations (version) VALUES ('20150807113927');

INSERT INTO schema_migrations (version) VALUES ('20150811135959');
```

This file is then used to produce a new `_test` database and will have `development` as environment and not `test`.